lib/grpc/stub.ex

```diff
@@ -131,13 +131,46 @@ defmodule GRPC.Stub do
   """
   @spec connect(String.t(), keyword()) :: {:ok, Channel.t()} | {:error, any()}
   def connect(addr, opts \\ []) when is_binary(addr) and is_list(opts) do
-    {host, port} =
-      case String.split(addr, ":") do
-        [host, port] -> {host, port}
-        [socket_path] -> {{:local, socket_path}, 0}
-      end
+    case URI.parse(addr) do
```

Comment:
```suggestion
    # This works because we only accept `http` and `https` schemes (allowlisted below explicitly)
    # addresses like "localhost:1234" parse as if `localhost` is the scheme for URI, and this falls through to
    # the base case. Accepting only `http/https` is a trait of `connect/3`.
    
    case URI.parse(addr) do
```